### PR TITLE
Timers for Shock

### PR DESCRIPTION
**Pull request by mcferront**
Enable by 
`#define SHOCK_TIMERS`

which is currently on in shock.h

### CODE REVIEW

File: src/burner/shock/shock.h [hunk 1]
```diff
@@ -9,10 +9,33 @@
 
 #define SHOCK_VERSION "1.0.2"
 
+typedef unsigned char byte;
```
Comment (jhawkzz) [inline src/burner/shock/shock.h:12]: Why this and not UINT8?

File: src/burner/shock/shockprofiler.h [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        mNode->val->Start();
+    }
+
+    static void EndScope()
+    {
+        mNode->val->Stop();
+        mNode = mNode->parent;
+    }
+
+    static void TraverseDepth(void* context, Tree<Timer*, 64>::TreeCb cb)
```
Comment (jhawkzz) [inline src/burner/shock/shockprofiler.h:67]: Is 64 max depth? Worth making a #define or nah?

File: src/burner/shock/ui/states/uibasestate.h [hunk 3]
```diff
@@ -34,7 +34,7 @@ enum UIState
 #define UI_CURSOR_X_OFFSET (52)  //The amount of pixels to decrement by when rendering the selection cursor
 #define MAX_MENU_ITEMS     (15)
 #define UI_COLOR_DISABLED  (0x4A49)
-#define UI_COLOR_ENABLED   (0XFFEA)
+#define UI_COLOR_ENABLED   (0XFFEAu)
```
Comment (jhawkzz) [inline src/burner/shock/ui/states/uibasestate.h:37]: Does u ensure unsigned?

**Review by jhawkzz — APPROVED** [review]
Looks super good--you'll have to walk me thru it at some point so I can learn

File: src/burner/shock/shock.h (hunk 1 above)
Comment (mcferront) [reply src/burner/shock/shock.h:12]: I just like using 'byte' it feels so nice, I can change it if you want

File: src/burner/shock/shockprofiler.h (hunk 2 above)
Comment (mcferront) [reply src/burner/shock/shockprofiler.h:67]: probably?  Let me get more timers in and we'll see

File: src/burner/shock/ui/states/uibasestate.h (hunk 3 above)
Comment (mcferront) [reply src/burner/shock/ui/states/uibasestate.h:37]: Yeah, it seems ridiculous but it fixes a lot of warnings on windows

**Pull request merged**